server/jetstream_cluster.go

```diff
@@ -2608,12 +2629,46 @@ func (mset *Stream) snapshot() []byte {
 
 // processClusteredMsg will propose the inbound message to the underlying raft group.
 func (mset *Stream) processClusteredInboundMsg(subject, reply string, hdr, msg []byte) error {
-	mset.mu.Lock()
 
 	// For possible error response.
 	var response []byte
-	canRespond := !mset.config.NoAck && len(reply) > 0 && mset.isLeader()
-	sendq := mset.sendq
+
+	mset.mu.RLock()
+	canRespond := !mset.config.NoAck && len(reply) > 0
```

Comment:
The `mst.isLeader()` condition disappeared. Is that intentional?
I understand that if all nodes can check limits, any can respond, but `canRespond` is not only used here, it is used also when processing the message. If all nodes reply, is that not wasting traffic?

Reply:
ok